### ISSUE TITLE
ENOENT error thrown when Prelude not completely imported

### ISSUE THREAD

**Issue by AlexaDeWit** [purescript-contrib/pulp#354]
https://github.com/AlexaDeWit/MySelfcareClient/blob/04f9e3774688e84374850ea08b640ba5e7f453f8/src/App.purs

This file in particular will result in the following build error...

```* Building project in /home/alexa/Workspace/my-selfcare-client
Compiling Main
fs.js:646
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: ENOENT: no such file or directory, open ''
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at /home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Node.FS.Sync/index.js:60:23
    at /home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Control.Monad.Eff/foreign.js:12:16
    at /home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Control.Monad.Eff/foreign.js:12:20
    at __do (/home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Main/index.js:559:173)
    at __do (/home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Main/index.js:564:23)
    at /home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Control.Monad.Eff/foreign.js:12:16
    at /home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Control.Monad.Eff/foreign.js:12:16
    at /home/alexa/.config/yarn/global/node_modules/purescript-psa/output/Control.Monad.Eff/foreign.js:12:16

```

If I import only Unit, unit from prelude.

The moment I import all of prelude however, the issue resolves. I'm not certain this bug is localised to pulp, but I figure this is the best place to begin.

This occurs for all pulp build chains, build, run, bundle, etc, and happens AFTER all compilation. (This is what I got when I -finished- fixing all my type errors)

**Comment by nwolverson** [comment]
That stack trace consists of entirely references to `purescript-psa`, have you tried using only `purescript-psa` directly to build? And compare to `pulp compile` ?

**Comment by AlexaDeWit** [comment]
I'm having trouble understanding how to use psa to perform a build, since it's just an error reporting frontend?

**Comment by roryc89** [comment]
I had a similar error. I only found it when using do notation and explicitly importing from Prelude, without all the imports required for do notation. 
`import Prelude (Unit, map, pure, unit, ($))`
It threw the error @AlexaDeWit mentioned instead of the usual message of "You're probably using do notation etc.".

**Comment by roryc89** [comment]
Actually I was using `join` as well and not importing it so that could have been the cause as well

**Comment by maddie927** [comment]
Also seeing this when using `do` without `Prelude` imported with `psa` installed. Uninstalling `psa` fixes it.

**Comment by hdgarrood** [comment]
I've just seen this too. A slightly easier workaround is to use the `--no-psa` flag. I'll see if anything similar has been reported on the `psa` repo.

**Comment by natefaubion** [comment]
Copied from the psa ticket:

> While psa should handle this gracefully, this is caused by  the compiler reporting a module and position information, but setting  the file to an empty string.

**Comment by natefaubion** [comment]
This was fixed with purescript-psa v0.7.3.

Associated purescript ticket https://github.com/purescript/purescript/issues/3418

**Comment by hdgarrood** [comment]
Thanks @natefaubion! To clarify for anyone running into this, `pulp` doesn't bundle `psa`; instead, it uses `psa` automatically whenever it can be found on your PATH. So if you're seeing this, all you need to do is update `psa`:

```
$ npm install [-g] purescript-psa
```
